### PR TITLE
seq-cli: init at 2024.3.922

### PR DESCRIPTION
**Pull request by hauskens**
Added seq-cli package, a CLI for seq log servers https://datalust.co/seq
Used nix init for boilerplate and specified the projectFile, since the repo seems to contain some sample code under `src/Roastery`  which does not seem to be related to the CLI.
Also added version test.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by hauskens** [comment]
Hi @NixOS/dotnet , this is my first time packaging a dotnet project, could you take a look and provide some feedback?

File: pkgs/by-name/se/seq-cli/deps.nix
Comment (GGG-KILLER) [inline pkgs/by-name/se/seq-cli/deps.nix:1]: The `deps.nix` file shouldn't be formatted.

File: pkgs/by-name/se/seq-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  lib,
+  buildDotnetModule,
+  fetchFromGitHub,
+  dotnetCorePackages,
+  testers,
+  seq-cli,
```
Comment (GGG-KILLER) [inline pkgs/by-name/se/seq-cli/package.nix:7]: Instead of adding the package to its own deps, use the function format of `buildDotnetModule`.

**Review by GGG-KILLER — CHANGES_REQUESTED** [review]
Other than these minor points, everything looks good.

File: pkgs/by-name/se/seq-cli/package.nix (hunk 1 above)
Comment (hauskens) [reply pkgs/by-name/se/seq-cli/package.nix]: Thanks for the input! I've applied them now as suggested

File: pkgs/by-name/se/seq-cli/deps.nix
Comment (hauskens) [reply pkgs/by-name/se/seq-cli/deps.nix:1]: I was not aware of that, however now that i ran `nix-build -A seq-cli.fetch-deps && ./result pkgs/by-name/se/seq-cli/deps.nix` the build seems to complain about formatting. Is there anything i can do about that failing check?

Comment (GGG-KILLER) [reply pkgs/by-name/se/seq-cli/deps.nix:1]: For now we've just been ignoring that error as the standard is to not format auto-generated files.

**Comment by GGG-KILLER** [comment]
Result of `nixpkgs-review pr 348595` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>seq-cli</li>
  </ul>
</details>

**Review by GGG-KILLER — APPROVED** [review]
LGTM

**Comment by Scrumplex** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 348595`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>seq-cli</li>
  </ul>
</details>

**Review by Scrumplex — APPROVED** [review]
Ignoring the nixfmt warning for `deps.nix` as it's autogenerated.

**Pull request merged**